server/src/routes/api/v1/dogParksRouter.js

```diff
@@ -44,4 +43,21 @@ dogParksRouter.get("/:id", async (req, res) => {
     }
 })
 
+dogParksRouter.delete("/:id", async (req, res) => {
+    const { id } = req.params
+    
+    try {
+        const park = await DogPark.query().findById(id)
+        const serializedPark = DogParksSerializer.detailsForShow(park)
+        const parkReviews = await park.$relatedQuery("reviews")
+        Promise.all(parkReviews.map( async(review) => {
+            await Review.query().deleteById(review.id)
+        }))
+        await DogPark.query().deleteById(id)
+        return res.status(200).json({ park: serializedPark })
```

Comment:
the database should do the heavy lifting work for us of deleting associated reviews through `onDelete("CASCADE")`. this can be added via a new migration:

```
exports.up = async (knex) => {
  return knex.schema.alterTable('reviews', table => {
    table.dropForeign("dogParksId")
    table.foreign("dogParksId").references("dogParks.id").onDelete("CASCADE")
  })
}

/**
 * @param {Knex} knex
 */
exports.down = (knex) => {
  return knex.schema.alterTable('reviews', table => {
    table.dropForeign("dogParksId")
    table.foreign("dogParksId").references("dogParks.id").onDelete("NO ACTION")
  })
}
```

this says whenever the referenced dogPark is deleted, also delete the associated reviews